cirq/google/engine/engine_sampler.py

```diff
@@ -61,6 +61,34 @@ def run_sweep(
                                          gate_set=self._gate_set)
         return job.results()
 
+    def run_batch(
+            self,
+            programs: List['cirq.Circuit'],
+            params_list: Optional[List['cirq.Sweepable']] = None,
+            repetitions: Union[int, List[int]] = 1,
+    ) -> List['cirq.TrialResult']:
```

Comment:
I am not sure if the sampler should flatten the results.  I think returning an List[List['cirq.TrialResult']] or maybe even an object would be better.

I realize that engine.results() only returns List['cirq.TrialResult'], but I was considering going back and adding a engine.batch_results() that returned an unflattened list.

Reply:
I agree that a list of lists is better, but yes, here I was just following the structure of job.results(). I'll change the Sampler interface to return a list of lists. For now, I could just un-flatten the list here, and when you implement batch_results we can change it to use that. Does that sound good? If you'll implement that soon I could also wait.

Reply:
I put out #3303 for this.  What do you think?  Can we use it here?

Reply:
Done.